File: visitor/oc/bytedesk-oc/ui/controllers/leavemsg/BDLeaveMessageViewController.m
//
//  BDLeaveMessageViewController.m
//  bytedesk-ui
//
//  Created by 萝卜丝 on 2019/4/8.
//  Copyright © 2019 bytedesk.com. All rights reserved.
//

#import "BDLeaveMessageViewController.h"
#import "BDLeaveHistoryViewController.h"
#import "BDVideoCompress.h"
#import "UITextView+Placeholder.h"

#import <AFNetworking/UIImageView+AFNetworking.h>
#import "BDCoreApis.h"
#import "BDSettings.h"
#import "BDUtils.h"
#import "BDUIApis.h"

#define DeviceHeight [[UIScreen mainScreen] bounds].size.height
#define DeviceWidth [[UIScreen mainScreen] bounds].size.width

@interface BDLeaveMessageViewController ()<UIImagePickerControllerDelegate, UINavigationControllerDelegate>

@property(nonatomic, strong) NSString *type;
@property(nonatomic, strong) NSString *uid;
//@property(nonatomic, strong) NSString *mAdminUid;
@property(nonatomic, strong) NSString *imageUrl;

@property(nonatomic, strong) UITextView *contentTextView;
@property(nonatomic, strong) UIImageView *imageView;
@property(nonatomic, strong) UITextField *mobileTextField;

@property(nonatomic, assign) BOOL mIsPush;
@property(nonatomic, assign) BOOL forceEnableBackGesture;

@property (nonatomic, strong) UIImagePickerController *mImagePickerController;

@end

@implementation BDLeaveMessageViewController

@synthesize contentTextView, imageView, mobileTextField;

- (void)initWithType:(NSString *)type withUid:(NSString *)uid withPush:(BOOL)isPush {
    NSLog(@"1.%s", __PRETTY_FUNCTION__);
    self.type = type;
    self.uid = uid;
    self.mIsPush = isPush;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"2.%s", __PRETTY_FUNCTION__);
    self.title = @"留言";
    self.view.backgroundColor = [UIColor systemGroupedBackgroundColor];
    if(@available(iOS 13.0,*)) {
        UIUserInterfaceStyle mode = UITraitCollection.currentTraitCollection.userInterfaceStyle;
        if (mode == UIUserInterfaceStyleDark) {
            self.view.backgroundColor = [UIColor blackColor];
        } else {
            self.view.backgroundColor = [UIColor whiteColor];
        }
    }
    //
    self.imageUrl = @"";
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"我的留言" style:UIBarButtonItemStylePlain target:self action:@selector(showLeaveRecords)];
    //
    if (self.mIsPush) {
        //
    }
    else {
        self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemClose target:self action:@selector(handleCloseButtonEvent:)];
        }
    self.forceEnableBackGesture = YES;// 当系统的返回按钮被屏蔽的时候，系统的手势返回也会跟着失效，所以这里要手动强制打开手势返回
    //
    UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyboard:)];
    singleTap.numberOfTapsRequired = 1;
    singleTap.numberOfTouchesRequired = 1;
    [self.view addGestureRecognizer:singleTap];
    
    // Dismiss the keyboard if user double taps on the background
    UITapGestureRecognizer *doubleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyboard:)];
    doubleTap.numberOfTapsRequired = 2;
    doubleTap.numberOfTouchesRequired = 1;
    [self.view addGestureRecognizer:doubleTap];
    //
    self.mImagePickerController = [[UIImagePickerController alloc] init];
    self.mImagePickerController.delegate = self;
    self.mImagePickerController.allowsEditing = YES;
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    NSLog(@"3.%s", __PRETTY_FUNCTION__);
}

- (void)viewWillLayoutSubviews {
    NSLog(@"4.%s", __PRETTY_FUNCTION__);
    [super viewWillLayoutSubviews];
    //
    contentTextView = [[UITextView alloc] initWithFrame:CGRectMake(10, 70, DeviceWidth - 20, 100)];
    contentTextView.font = [UIFont systemFontOfSize:16];
    contentTextView.placeholder = @"请输入留言内容";
    contentTextView.layer.cornerRadius = 5.0;
    contentTextView.layer.masksToBounds = YES;
    //    contentTextView.layer.borderColor = [UIColor colorWithWhite:0.0 alpha:0.2].CGColor;
    contentTextView.layer.borderColor = [UIColor systemGrayColor].CGColor;
    contentTextView.layer.borderWidth = 0.5;
    [self.view addSubview:contentTextView];
    //
    imageView = [[UIImageView alloc] initWithFrame:CGRectMake(10, 180, 100, 100)];
    //    [imageView setImage:[UIImage systemImageNamed:@"plus"]];
    [imageView setImage:[UIImage imageNamed:@"AlbumAddBtn.png" inBundle:[NSBundle bundleForClass:self.class] compatibleWithTraitCollection:nil]];
    imageView.userInteractionEnabled = YES;
    imageView.layer.cornerRadius = 5.0;
    imageView.layer.masksToBounds = YES;
    //    imageView.layer.borderColor = [UIColor colorWithWhite:0.0 alpha:0.2].CGColor;
    imageView.layer.borderColor = [UIColor systemGrayColor].CGColor;
    imageView.layer.borderWidth = 0.5;
    [self.view addSubview:imageView];
    //
    UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(presentViewController:)];
    singleTap.numberOfTapsRequired = 1;
    singleTap.numberOfTouchesRequired = 1;
    [imageView addGestureRecognizer:singleTap];
    //
    mobileTextField = [[UITextField alloc] initWithFrame:CGRectMake(10, 290, DeviceWidth - 20, 40)];
    mobileTextField.placeholder = @"请输入联系方式";
    mobileTextField.layer.cornerRadius = 5.0;
    mobileTextField.layer.masksToBounds = YES;
    //    contactTextField.layer.borderColor = [UIColor colorWithWhite:0.0 alpha:0.2].CGColor;
    mobileTextField.layer.borderColor = [UIColor systemGrayColor].CGColor;
    mobileTextField.layer.borderWidth = 0.5;
    [self.view addSubview:mobileTextField];
    //
    UIButton *submitButton = [[UIButton alloc] initWithFrame:CGRectMake(10, 340, DeviceWidth - 20, 40)];
    [submitButton setBackgroundColor:[UIColor systemOrangeColor]];
    [submitButton setTitle:@"提交留言" forState:UIControlStateNormal];
    [submitButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [submitButton addTarget:self action:@selector(submitButtonClicked:) forControlEvents:UIControlEventTouchUpInside];
    submitButton.layer.cornerRadius = 20.0;
    submitButton.layer.masksToBounds = YES;
    //    submitButton.layer.borderColor = [UIColor colorWithWhite:0.0 alpha:0.2].CGColor;
    submitButton.layer.borderColor = [UIColor systemGrayColor].CGColor;
    submitButton.layer.borderWidth = 0.5;
    [self.view addSubview:submitButton];
    //
    //    [self initRobotView];
}

#pragma mark -

- (void)submitButtonClicked:(id)sender {
    
    NSString *content = contentTextView.text;
    NSString *mobile = mobileTextField.text;
    NSLog(@"%s, content=%@, contact=%@", __PRETTY_FUNCTION__, content, mobile);
    
    if ([content stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]].length == 0 ||
        [mobile stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]].length == 0) {
        [BDUIApis showTipWithVC:self withMessage:@"留言内容 或 联系方式不能为空"];
        return;
    }
    
    [BDUIApis showTipWithVC:self withMessage:@"提交留言中，请稍后"];
    __weak __typeof(self)weakSelf = self;
    [BDCoreApis leaveMessage:self.type
                     withUid:self.uid
                 withMobile: mobile
                 withContent: content
                withImageUrl:self.imageUrl
               resultSuccess:^(NSDictionary *dict) {
        NSString *message = [dict objectForKey:@"message"];
        NSNumber *status_code = [dict objectForKey:@"status_code"];
        NSLog(@"%s message:%@, status_code:%@", __PRETTY_FUNCTION__, message, status_code);
        
        dispatch_async(dispatch_get_main_queue(), ^{
            if ([status_code isEqualToNumber:[NSNumber numberWithInt:200]]) {
                [BDUIApis showTipWithVC:weakSelf withMessage:message];
                //
                [weakSelf closeVC];
            } else {
                [BDUIApis showTipWithVC:weakSelf withMessage:message];
            }
        });
        
    } resultFailed:^(NSError *error) {
        [BDUIApis showErrorWithVC:weakSelf withMessage:error.localizedDescription];
    }];
}

- (void)closeVC {
    //
    [self dismissViewControllerAnimated:YES completion:^{
        
    }];
}

- (void)showLeaveRecords {
    BDLeaveHistoryViewController *bdleaveRecordVC = [[BDLeaveHistoryViewController alloc] init];
    [self.navigationController pushViewController:bdleaveRecordVC animated:YES];
}

- (BOOL)forceEnableInteractivePopGestureRecognizer {
    return self.forceEnableBackGesture;
}

// 针对Present打开模式，左上角返回按钮处理action
- (void)handleCloseButtonEvent:(id)sender {
    NSLog(@"%s", __PRETTY_FUNCTION__);
//    self.mIsViewControllerClosed = YES;
    [self.navigationController dismissViewControllerAnimated:YES completion:^{
    }];
}

// 针对Push打开模式，左上角返回按钮处理action
- (void)handleBackButtonEvent:(id)sender {
    NSLog(@"%s", __PRETTY_FUNCTION__);
//    self.mIsViewControllerClosed = YES;
    [self.navigationController popViewControllerAnimated:YES];
}


#pragma mark - UIScrollViewDelegate

-(void)scrollViewWillBeginDragging:(UIScrollView *)scrollView {
//    NSLog(@"%s", __PRETTY_FUNCTION__);
    [self.view endEditing:YES];
}

#pragma mark - TouchGestures

-(void)dismissKeyboard:(id)sender {
    [self.view endEditing:YES];
}

#pragma mark - KFFormSingleImageViewCellDelegate

- (void)presentViewController:(id)sender {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    
    [self authorizationPresentAlbumViewController];
}

- (void)authorizationPresentAlbumViewController {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    
//    if ([QMUIAssetsManager authorizationStatus] == QMUIAssetAuthorizationStatusNotDetermined) {
//        [QMUIAssetsManager requestAuthorization:^(QMUIAssetAuthorizationStatus status) {
//            dispatch_async(dispatch_get_main_queue(), ^{
//                [self presentAlbumViewController];
//            });
//        }];
//    } else {
        [self presentAlbumViewController];
//    }
}

- (void)presentAlbumViewController {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    
    self.mImagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
    [self presentViewController:self.mImagePickerController animated:YES completion:nil];
}

#pragma mark UIImagePickerControllerDelegate

- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    [picker dismissViewControllerAnimated:YES completion:^{
        [self performSelector:@selector(dealWithImage:) withObject:info];
    }];
}

- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    [picker dismissViewControllerAnimated:YES completion:nil];
}

- (void)dealWithImage:(NSDictionary *)info {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    //
    NSString *mediaType = [info objectForKey:UIImagePickerControllerMediaType];
    if([mediaType isEqualToString:@"public.movie"]) {
        NSLog(@"%s 1", __PRETTY_FUNCTION__);
        //被选中的是视频
        //        // [QMUITips showError:@"请选择图片" inView:self.view hideAfterDelay:2];
        // TODO：上传视频
        NSURL *videoURL = [info objectForKey:UIImagePickerControllerMediaURL];
        NSLog(@"videoURL %@", [videoURL absoluteString]);

    } else if([mediaType isEqualToString:@"public.image"]) {
        NSLog(@"%s 2", __PRETTY_FUNCTION__);
        //获取照片实例
        UIImage *image = [info objectForKey:UIImagePickerControllerOriginalImage];
        UIImageOrientation imageOrientation = image.imageOrientation;
        if (imageOrientation != UIImageOrientationUp) {
            UIGraphicsBeginImageContext(image.size);
            [image drawInRect:CGRectMake(0, 0, image.size.width, image.size.height)];
            image = UIGraphicsGetImageFromCurrentImageContext();
            UIGraphicsEndImageContext();
        }
        NSData *imageData = UIImageJPEGRepresentation(image, 0.6); // 压缩
        [self uploadImageData:imageData];
    }
}

- (void)uploadImageData:(NSData *)imageData {
    //
    [BDUIApis showTipWithVC:self withMessage:@"图片上传中"];
    NSString *localId = [[NSUUID UUID] UUIDString];
    NSString *imageName = [NSString stringWithFormat:@"%@_%@.png", [BDSettings getUsername], [BDUtils getCurrentTimeString]];
    [BDCoreApis uploadImageData:imageData withImageName:imageName withLocalId:localId resultSuccess:^(NSDictionary *dict) {
        //
        NSNumber *status_code = [dict objectForKey:@"status_code"];
        if ([status_code isEqualToNumber:[NSNumber numberWithInt:200]]) {
            
            self.imageUrl = dict[@"data"];
            [self.imageView setImageWithURL:[NSURL URLWithString:self.imageUrl] placeholderImage:[UIImage systemImageNamed:@"plus"]];
//            self.mFileUrl = imageUrl;
            NSLog(@"imageUrl %@", self.imageUrl);
            
        } else {
            // [QMUITips showError:@"上传图片错误" inView:self.view hideAfterDelay:2];
        }
    } resultFailed:^(NSError *error) {
        
    }];
    
}

@end

File: visitor/swift/Sources/bytedesk-swift/ViewController/Leavemsg/BDLeaveMessageViewController.swift
//
//  File.swift
//  
//
//  Created by 宁金鹏 on 2023/9/12.
//

import UIKit

class BDLeaveMessageViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
    
    var type: String?
    var uid: String?
    // var mAdminUid: String?
    var imageUrl: String?

    var contentTextView: UITextView?
    var imageView: UIImageView?
    var mobileTextField: UITextField?

    var mIsPush: Bool = false
    var forceEnableBackGesture: Bool = false

    var mImagePickerController: UIImagePickerController?
    
    let deviceHeight = UIScreen.main.bounds.size.height
    let deviceWidth = UIScreen.main.bounds.size.width
    
    public func initWithType(type: String, uid: String, isPush: Bool) {
        NSLog("1.%@", #function)
        self.type = type
        self.uid = uid
        self.mIsPush = isPush
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        NSLog("2.%@", #function)
        self.title = "留言"
        self.view.backgroundColor = UIColor.systemGroupedBackground
        if #available(iOS 13.0, *) {
            let mode = UITraitCollection.current.userInterfaceStyle
            if mode == .dark {
                self.view.backgroundColor = UIColor.black
            } else {
                self.view.backgroundColor = UIColor.white
            }
        }
        self.imageUrl = ""
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "我的留言", style: .plain, target: self, action: #selector(showLeaveRecords))
        if !self.mIsPush {
            self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .close, target: self, action: #selector(handleCloseButtonEvent))
        }
        self.forceEnableBackGesture = true
        
        let singleTap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
        singleTap.numberOfTapsRequired = 1
        singleTap.numberOfTouchesRequired = 1
        self.view.addGestureRecognizer(singleTap)
        
        let doubleTap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
        doubleTap.numberOfTapsRequired = 2
        doubleTap.numberOfTouchesRequired = 1
        self.view.addGestureRecognizer(doubleTap)
        
        self.mImagePickerController = UIImagePickerController()
        self.mImagePickerController!.delegate = self
        self.mImagePickerController!.allowsEditing = true
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        NSLog("3.%@", #function)
    }

    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        NSLog("4.%@", #function)
        contentTextView = UITextView(frame: CGRect(x: 10, y: 70, width: deviceWidth - 20, height: 100))
        contentTextView!.font = UIFont.systemFont(ofSize: 16)
//        contentTextView!.placeholder = "请输入留言内容"
        contentTextView!.layer.cornerRadius = 5.0
        contentTextView!.layer.masksToBounds = true
        contentTextView!.layer.borderColor = UIColor.systemGray.cgColor
        contentTextView!.layer.borderWidth = 0.5
        self.view.addSubview(contentTextView!)
        
        imageView = UIImageView(frame: CGRect(x: 10, y: 180, width: 100, height: 100))
        imageView!.image = UIImage(named: "AlbumAddBtn.png", in: .module, with: nil)
        imageView!.isUserInteractionEnabled = true
        imageView!.layer.cornerRadius = 5.0
        imageView!.layer.masksToBounds = true
        imageView!.layer.borderColor = UIColor.systemGray.cgColor
        imageView!.layer.borderWidth = 0.5
        self.view.addSubview(imageView!)
        //
        let singleTap = UITapGestureRecognizer(target: self, action: #selector(presentViewController(_:)))
        singleTap.numberOfTapsRequired = 1
        singleTap.numberOfTouchesRequired = 1
        imageView!.addGestureRecognizer(singleTap)
        
        mobileTextField = UITextField(frame: CGRect(x: 10, y: 290, width: deviceWidth - 20, height: 40))
        mobileTextField!.placeholder = "请输入联系方式"
        mobileTextField!.layer.cornerRadius = 5.0
        mobileTextField!.layer.masksToBounds = true
        mobileTextField!.layer.borderColor = UIColor.systemGray.cgColor
        mobileTextField!.layer.borderWidth = 0.5
        self.view.addSubview(mobileTextField!)
        
        let submitButton = UIButton(frame: CGRect(x: 10, y: 340, width: deviceWidth - 20, height: 40))
        submitButton.backgroundColor = UIColor.systemOrange
        submitButton.setTitle("提交留言", for: .normal)
        submitButton.setTitleColor(UIColor.white, for: .normal)
        submitButton.addTarget(self, action: #selector(submitButtonClicked), for: .touchUpInside)
        submitButton.layer.cornerRadius = 20.0
        submitButton.layer.masksToBounds = true
        submitButton.layer.borderColor = UIColor.systemGray.cgColor
        submitButton.layer.borderWidth = 0.5
        self.view.addSubview(submitButton)
    }
    
    @objc func submitButtonClicked() {
        let content = contentTextView?.text
        let mobile = mobileTextField?.text
        
        if (content!.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines).isEmpty ||
            mobile!.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines).isEmpty) {
            BDToast.show(message: "留言内容或手机号不能为空")
            return
        }
        
        BDUIApis.showTip(with: self, message: "留言中，请稍后")
        BDCoreApis.createLeaveMessage(type: type,
                                uid: uid,
                                mobile: mobile,
                                content: content,
                                imageUrl: imageUrl) { leaveMsgResult in
            BDToast.show(message: leaveMsgResult.message!)
            self.navigationController?.dismiss(animated: true)
        } onFailure: { error in
            BDToast.show(message: error)
        }
    }
    
    @objc func closeVC() {
        self.dismiss(animated: true)
    }
    
    @objc func showLeaveRecords() {
        let bdleaveRecordVC = BDLeaveHistoryViewController()
        self.navigationController?.pushViewController(bdleaveRecordVC, animated: true)
    }
    
    @objc func handleCloseButtonEvent() {
        self.navigationController?.dismiss(animated: true)
    }
    
    @objc func handleBackButtonEvent() {
        self.navigationController?.popViewController(animated: true)
    }
    
    @objc func dismissKeyboard() {
        self.view.endEditing(true)
    }
    
    // MARK:
    
    @objc func presentViewController(_ sender: Any) {
        print("1.\(#function)")
        
        authorizationPresentAlbumViewController()
    }

    func authorizationPresentAlbumViewController() {
        print("2.\(#function)")
        presentAlbumViewController()
    }

    func presentAlbumViewController() {
        print("3.\(#function)")
        
        mImagePickerController!.sourceType = .photoLibrary
        present(mImagePickerController!, animated: true, completion: nil)
    }

    // MARK: - UIImagePickerControllerDelegate

    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
        print("4.\(#function)")
        picker.dismiss(animated: true) {
            self.perform(#selector(self.dealWithImage(_:)), with: info)
        }
    }

    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
        print("5.\(#function)")
        picker.dismiss(animated: true, completion: nil)
    }

    @objc func dealWithImage(_ info: [UIImagePickerController.InfoKey: Any]) {
        print("6.\(#function)")
        let mediaType = info[.mediaType] as? String
        if mediaType == "public.movie" {
            print("6.1.\(#function)")
            // Selected media is a video
            // [QMUITips showError:@"请选择图片" inView:self.view hideAfterDelay:2];
            // TODO: Upload video
            if let videoURL = info[.mediaURL] as? URL {
                print("videoURL", videoURL.absoluteString)
                
                // [BDVideoCompress compressVideoWithVideoUrl:videoURL withBiteRate:@(1500 * 1024) withFrameRate:@(30) withVideoWidth:@(960) withVideoHeight:@(540) compressComplete:^(id responseObjc) {
                //     //
                //     NSString *filePathStr = [responseObjc objectForKey:@"urlStr"];
                //     NSURL *compressvideourl = [NSURL fileURLWithPath:filePathStr];
                //     //
                //     NSData *videoData = [NSData dataWithContentsOfURL:compressvideourl];
                //     [self uploadVideoData:videoData];
                // }];
            }
        } else if mediaType == "public.image" {
            print("6.2.\(#function)")
            // Get the selected photo
            if var image = info[.originalImage] as? UIImage {
                var imageOrientation = image.imageOrientation
                if imageOrientation != .up {
                    UIGraphicsBeginImageContext(image.size)
                    image.draw(in: CGRect(x: 0, y: 0, width: image.size.width, height: image.size.height))
                    image = UIGraphicsGetImageFromCurrentImageContext() ?? image
                    UIGraphicsEndImageContext()
                }
                if let imageData = image.jpegData(compressionQuality: 0.6) {
                    uploadImageData(imageData)
                }
            }
        }
    }

    func uploadImageData(_ imageData: Data) {
        print("7.\(#function)")
        
        // [BDUIApis showTipWithVC:self withMessage:@"图片上传中"];
        let localId = UUID().uuidString
//        let imageName = "\(BDSettings.getUsername())_\(BDUtils.getCurrentTimeString()).png"
//        BDCoreApis.uploadImageData(imageData, withImageName: imageName, withLocalId: localId, resultSuccess: { dict in
//            print("\(#function), uploadImageData: \(dict)")
//            if let statusCode = dict["status_code"] as? NSNumber, statusCode.intValue == 200 {
//                if let imageUrl = dict["data"] as? String {
//                    self.mImageUrl = imageUrl
//                    self.imageView?.setImage(with: URL(string: self.mImageUrl), placeholder: UIImage(named: "AlbumAddBtn.png", in: Bundle(for: Self.self), compatibleWith: nil))
//                    // self.mFileUrl = imageUrl;
//                    print("imageUrl", self.mImageUrl)
//                } else {
//                    // [BDUIApis showErrorWithVC:self withMessage:@"上传图片错误"];
//                }
//            }
//        }, resultFailed: { error in
//            if let error = error {
//                // [BDUIApis showErrorWithVC:self withMessage:error.localizedDescription];
//            }
//        })
    }
    
}
